### DISCUSSION TITLE
Offer to help maintain

### DISCUSSION THREAD

**Discussion by lourd** [redux-saga/redux-saga#2299] [General]
Hey @Andarist, @neurosnap, et al,

First off, I'd like to acknowledge and thank you for your long term commitment of maintaining this library, Mateusz. Open source work is difficult and often thankless, and the many years you've been doing it essentially solo for this library after the original author and maintainers moved on is really impressive. So for that, thank you.

With that said, I think we can acknowledge that the project is in need of some good attention. It's been 3 years since the last release. Making contributions is difficult and slow. PRs with beneficial contributions sat open for months and years. One such example (https://github.com/redux-saga/redux-saga/pull/2004) was finally merged a few weeks ago, but a new version of the library still hasn't been published with the changes. The PRs I've made recently (#2295, #2296, #2297) that I considered small and strictly beneficial, i.e. no breaking changes, either haven't been reviewed or got stuck in discussion about the tradeoff of making even a tiny tweak to the library's public API.

State of contributing aside, the ecosystem has continued to evolved and the library hasn't kept pace. There's several disparate pieces needed to use this library effectively in a production capacity — typed-redux-saga + eslint-plugin-typed-redux-saga, redux-saga-test-plan — and the path to finding those pieces is not clear. And there's the still-missing piece of using this library in hooks-based React applications, applications that manage their state with `useReducer` instead of an external Redux store.

I'd like to offer to help fix these issues by becoming a maintainer. I'm a long-time user of the library, stretching back to 2015, and am using it daily at my current job. I've architected our web application to use sagas really effectively, and I'd like to contribute key pieces of that back to the library. This includes a robust `useSaga` hook implementation.

Please let me know your thoughts. I'd be happy to schedule a video call so we can meet, discuss the project, and y'all can verify this isn't some elaborate supply chain attack 🙃 

Cheers,
Lou

**Comment by Andarist** [comment]
Thanks for the offer - it is really appreciated! It might be best to schedule a call to just get to know each other and discuss some things.

I would especially like to discuss the implementation of `useSaga` that you are talking about cause I think that it's almost not possible for it to support React 18 and its strict effects. I'd like to learn more about how you are using it etc because this could inform Dan about the real need to introduce some new APIs/options in React itself.
